FWCore/Catalog/src/InputFileCatalog.cc

```diff
@@ -76,30 +80,42 @@ namespace edm {
       std::vector<std::string> const& tmp_dataCatalogs = localconfservice->trivialDataCatalogs();
       if (!fileLocators_trivalCatalog_.empty())
         fileLocators_trivalCatalog_.clear();
-
-      //require the first file locator to success so obvious mistakes in data catalogs, typos for example, can be catched early. Note that tmp_dataCatalogs is not empty at this point. The protection is done inside the trivialDataCatalogs() of SiteLocalConfigService
-      fileLocators_trivalCatalog_.push_back(std::make_unique<FileLocator>(tmp_dataCatalogs.front()));
-
-      for (auto it = tmp_dataCatalogs.begin() + 1; it != tmp_dataCatalogs.end(); ++it) {
+      //Construct all file locators from data catalogs. If a data catalog is invalid (wrong protocol for example), it is skipped and no file locator is constructed (an exception is thrown out from FileLocator::init).
+      for (auto it = tmp_dataCatalogs.begin(); it != tmp_dataCatalogs.end(); ++it) {
```

Comment:
Now this loop (and the similar loop below) could become range-for along
```suggestion
      for (auto const& catalog : tmp_dataCatalogs) {
```